LongWarOfTheChosen/Src/LW_Overhaul/Classes/UIMission_LWLaunchDelayedMission.uc

```diff
@@ -621,6 +699,122 @@ simulated function XComGameState_LWAlienActivity GetActivity()
+// It enables and disables navigation for various components, then ultimately kills the screen's navigation system with Navigator.Clear()
+// before adding 3 buttons back into it. Furthermore, Long War 2 calls UIUtilities_LW --> BuildMissionInfoPanel() after this function,
+// and adds a problematic UIPanel, MissionExpiryPanel, to the navigation system. The goal is to clean everything up and make it controller compatible.
+//
+// Please note that :
+// 1.] UIMission --> RefreshNavigation() sometimes refers to LockedPanel; however, LockedPanel is never used in this class; therefore, code referring to it has been removed.
+// 2.] UIUtilities_LW --> BuildMissionInfoPanel() has been modified so that it no longer adds a problematic UIPanel to the navigation system.
+simulated function RefreshNavigation()
+{
+	// KDM : Start with an empty navigation system.
+	Navigator.Clear();
+	Navigator.LoopSelection = true;
+
+	// KDM : There are 5 potentially selectable buttons : Button1, Button2, Button3, ConfirmButton, and IgnoreButton.
+	// If a given button is visible then it will be used, as per a WOTC comment in UIMission --> BuildScreen(); therfore, add it to the navigation system.
+	// If a given button is not visible it won't be used so just remove it; I am uncertain why code within UIMission --> RefreshNavigation() hides
+	// the button before removing it, as this appears unnecessary.
+	AddToNavigatorOrRemove(Button1);
+	AddToNavigatorOrRemove(Button2);
+	AddToNavigatorOrRemove(Button3);
+	AddToNavigatorOrRemove(ConfirmButton);
+	AddToNavigatorOrRemove(IgnoreButton);
+
+	// KDM : At this point, our navigation system is set up. We now need to determine which button to select initially.
+	// A reasonable ordering, taking into account button visibility, is : ConfirmButton --> Button1 --> Button2 --> Button3 --> IgnoreButton 
+	if (ConfirmButton.bIsVisible)
+	{
+		Navigator.SetSelected(ConfirmButton);
+	}
+	else if (Button1.bIsVisible)
+	{
+		Navigator.SetSelected(Button1);
+	}
+	else if (Button2.bIsVisible)
+	{
+		Navigator.SetSelected(Button2);
+	}
+	else if (Button3.bIsVisible)
+	{
+		Navigator.SetSelected(Button3);
+	}
+	else if (IgnoreButton.bIsVisible)
+	{
+		Navigator.SetSelected(IgnoreButton);
+	}
+}
+
+// Part of the fix for issue #961. This function makes sure that the "BOOST INFILTRATION", "VIEW SQUAD", "ABORT", "LAUNCH MISSION", and "WAIT"
+// buttons can all be "clicked" via the controller's A button or the Enter/Spacebar keys.
+simulated function bool OnUnrealCommand(int cmd, int arg)
+{
+	local bool bHandled;
+	local UIButton SelectedButton;
+
+	if (!CheckInputIsReleaseOrDirectionRepeat(cmd, arg))
+	{
+		return false;
+	}
+
+	bHandled = true;
+
+	switch(cmd)
+	{
+		// KDM : A button clicks on the selected button, if such a button exists.
+		// If no button was selected then fall down to the next case; this is to stay consistent with UIMission --> OnUnrealCommand().
```

Comment:
Wait, doesn't this mean that Enter, Space, and A will close the screen if there is no button selected? I'm assuming that never happens, but it's weird behaviour for sure.

Reply:
Yep - That's why I added the comment to make sure I knew why this was done. In UIMission, if neither the confirm button nor Button1 are clicked, it just moves down and exits the screen if CanBackOut() is true.

The exact comment in UIMission is : "If you don't have a current button, fall down and hit the Navigation system". Not sure exactly when that would occur, but wanted to be consistent with it. If issues arise it is easy to add a break.